src/Microsoft.Extensions.DependencyInjection/ServiceProvider.cs

```diff
@@ -186,7 +186,7 @@ public void Dispose()
             }
         }
 
-        internal object CaptureDisposable(object service)
+        internal virtual object CaptureDisposable(object service)
```

Comment:
To clarify, it would be worth testing if you could mock Dispose in fake services in your tests, but since dispose isn't called either way, at worst we regress perf. We should have real perf tests instead of testing a call (which you just made virtual ironically) gets elided.